src/controllers/FieldsController.php

```diff
@@ -360,6 +360,11 @@ public function actionSaveField(): ?Response
             $fieldUid = null;
         }
 
+        $settings = $this->request->getBodyParam('types.' . $type);
+        if (isset($oldField) && property_exists($oldField, 'maxRowId')) {
+            $settings['maxRowId'] = $oldField->maxRowId;
+        }
+
```

Comment:
FieldsController should not be giving special treatment to any one field type like this.